### PR TITLE
Introduced abstract base class for infection model + a nasty bug fix

### PR DESCRIPTION
**Pull request by ghost**
Hi,
since we're experimenting with different infection models, I've removed the infection code from HumanBase and created an abstract base class `AbstractInfection`. The existing, probability-based infection has been moved to `ProbabilityBasedInfection.cs`, all in the *Infection* folder. The infection model can simply be attached to the Human as another component, as I already did to the prefabs.

Also, Unity Script Execution order sometimes caused the End Game Controller to finish the game before it has even started. I solved this by introducing the `EndGameController.InitComplete()` method. The controller's update method will now do nothing before `InitComplete` is called.

@maccxs If you've already written code for your SEIR model, could you also move it to a class derived from `AbstractInfection`? This way, switching between models becomes most easy.

### CODE REVIEW

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs [hunk 1]
```diff
@@ -309,5 +238,6 @@ public void UpdateSpriteImage()
             }
         }
 
+
```
Comment (holke) [inline YouVsVirus/Assets/Scripts/Components/HumanBase.cs]: accidental newline?

**Review by holke — COMMENTED** [review]
On the first look this seems good to me.
It would have been better to seperate the bug fix from the abstract base class in two pull requests, since they are unrelated.
I think @maccxs  should look at this and make the final approval, since she is more familiar with the infection model.

File: YouVsVirus/Assets/Scripts/Components/HumanBase.cs (hunk 1 above)
Comment (ghost) [reply YouVsVirus/Assets/Scripts/Components/HumanBase.cs]: probably.

**Review by maccxs — APPROVED** [review]
This looks good and ready to be merged. Thanks for the work.

**Pull request merged**